transforms/transforms_test.go

```diff
@@ -362,6 +363,45 @@ func TestUniqueErrorForInvalidHeader(t *testing.T) {
 	}
 }
 
+func TestGroupBy(t *testing.T) {
+	transform := GroupBy(KeyIdentifier("group"))
+	input := []optimus.Row{
+		{"group": "1", "key": "a"},
+		{"group": "2", "key": "d"},
+		{"group": "3", "key": "g"},
+		{"group": "1", "key": 1},
+		{"group": "2", "key": 2},
+		{"group": "3", "key": 3},
+		{"group": "1", "key": 'b'},
+		{"group": "2", "key": 'e'},
+		{"group": "3", "key": 'h'},
+	}
+	expected := []optimus.Row{
+		{"1": []optimus.Row{input[0], input[3], input[6]}},
```

Comment:
That makes a lot of sense.  I agree it's super awkward to have one, variable key.  I'd go with something like any one of those, with a preference for `{"value": "1", "rows": []optimus.Row{...}}` but really any naming of the keys is fine.